lib/request.js

```diff
@@ -769,7 +769,9 @@ AWS.Request.addPromisesToClass = function addPromisesToClass(PromiseDependency)
         if (resp.error) {
           reject(resp.error);
         } else {
-          resolve(resp.data);
+          var data = AWS.util.copy(resp.data);
```

Comment:
AWS.util.copy is a shallow object clone, so this would just create a new object and assign the members from the output shape structure to that new object. Doing the copy prevents a reference cycle of `data.$response.data.$response.data...` etc.